includes/ndef.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "esphome.h"
+
+#include "NfcAdapter.h"
+#include "PN532/PN532.h"
+#include "PN532_SPI/PN532_SPI.h"
+
+static const char *TAG = "ndef_tag_reader";
+
+static const std::string TAG_PREFIX = "home-assistant.io/tag/";
+
+class NDEFTagReader : public PollingComponent, public TextSensor {
+ public:
+  NDEFTagReader() : PollingComponent(1000) {}
+  void setup() override {
+    this->pn532spi_ = new PN532_SPI(SPI, 0);
+    this->nfc_ = new NfcAdapter(*this->pn532spi_);
+    this->nfc_->begin();
+  }
+
+  void update() override {
+    if (this->nfc_->tagPresent()) {
```

Comment:
Let's rewrite to use a guard clause. So if not present: return. That way code is easier to read.

Reply:
Can do this